### DISCUSSION TITLE
When to show notifications for macOS app?

### DISCUSSION THREAD

**Discussion by jaanus** [tact/public#98] [General]
Here’s an interesting change request: #97. In a nutshell:

> Do not show notifications when app is closed

There are at least two ways how notifications can behave on macOS:

* Always show when they arrive (currently done in Tact)
* Only show when the app is running

Let’s hear pros and cons for both. I’m open to changing it in Tact if there is good motivation for it.

Note also that you can currently mute chats in Tact per each chat, or also turn off notifications globally in system settings on macOS or iOS.

**Comment by jenskasemets** [comment]
Tact is only chat app, out of ca 10 I use, that shows notifications when app is not running. It is counter-productive, because I want to control when app interact with me or not. E.g. I do not want notification with private messages to appear during presentations, then I close all the apps. And then suddenly in the middle of the presenation Tact notifications start to appear when only one app is running on whole computer.
This forces me to disable notifications for the whole app, which does not sound also right.

**Reply by jaanus** [reply]
For the “do not show notifications during presentations” case, I myself use “Do Not Disturb or Focus or whatever it is called today”. Even if I quit all the apps, there’s still some system notifications that may appear at random times.

<img width="159" alt="Screenshot 2021-12-20 at 09 42 25" src="https://user-images.githubusercontent.com/184253/146730452-f7382d6d-ad23-4073-9b3b-c8407a5fe3c7.png">

**Reply by priiduzilmer** [reply]
I tend to agree with Jens on this one. I think it is fair to not show you anything when you have fully and completely closed the app. Sort of goes with the basic principles of what you would expect from a correctly done app on macOS.

**Comment by jaanus** [comment]
This is now done in 1.2 (165). On macOS, the notifications appear only when the app is running.
